nautobot_floor_plan/forms.py

```diff
@@ -197,6 +197,26 @@ def clean_y_origin_seed(self):
         """Validate the Y origin seed."""
         return self._clean_origin_seed("y_origin_seed", "Y")
 
+    def clean(self):
+        """Custom clean method to validate floor plan dimensions."""
+        cleaned_data = super().clean()
+        x_size = self.cleaned_data.get("x_size")
+        y_size = self.cleaned_data.get("y_size")
+
+        # Get the configured limits
+        x_size_limit = get_app_settings_or_config("nautobot_floor_plan", "x_size_limit")
+        y_size_limit = get_app_settings_or_config("nautobot_floor_plan", "y_size_limit")
+
+        # Validate X size only if a limit is set
+        if x_size_limit is not None and x_size is not None and x_size > x_size_limit:
+            self.add_error("x_size", f"X size cannot exceed {x_size_limit}.")
```

Comment:
Can you add something to this error that explains that the limit is set in the config?